force-app/main/default/classes/AttendanceService.cls

```diff
@@ -212,22 +342,40 @@ public with sharing class AttendanceService {
                     relationshipName
                 )
             ) {
-                SObject targetObject = rollupTargetObjectById.get(parentRecord.Id);
-                if (targetObject == null) {
-                    targetObject = parentRecord;
-                    rollupTargetObjectById.put(targetObject.Id, targetObject);
-                }
-
-                List<ServiceDelivery__c> serviceDeliveries = (List<ServiceDelivery__c>) parentRecord.getSObjects(
-                    relationshipName
+                calculateLastServiceDeliveryRollups(
+                    parentRecord,
+                    relationshipName,
+                    lastServiceDateField
                 );
-                if (serviceDeliveries == null || serviceDeliveries.isEmpty()) {
-                    targetObject.put(lastServiceDateField, null);
-                } else {
+            }
+        }
+    }
+
+    private void calculateLastServiceDeliveryRollups(
```

Comment:
I like this, but adjusting to `setAttendanceFields` for consistency and because we aren't actually incrementing them in place -- we tally them from scratch every time.